assert/assertions.go

```diff
@@ -1936,6 +1942,9 @@ func EventuallyWithT(t TestingT, condition func(collect *CollectT), waitFor time
 				return true
 			}
 			tick = ticker.C
+		case <-failed:
+			collect.Copy(t)
+			return Fail(t, "Require assertion failed", msgAndArgs...)
```

Comment:
Hi @pducolin!

Unfortunately it doesn't fix the compatibility with `require` package. The return is too early so I cannot for example write `require.Equal(...)` which fails several but eventually is true.

I proposed an alternative solution https://github.com/stretchr/testify/pull/1481

Reply:
Interesting, I interpreted `require` as assertions that should never happen, like network errors. So I expect `require` to panic and stop the loop.

Reply:
I see `assert` and `require` as two different styles of writing tests with `testify`. I've seen projects that forbid a use of `assert` API and `require` is all they use.

As of network errors they also may be temporary. Suppose we dial a `127.0.0.1:30000` but the service is starting in background. So instead of:

```go
	require.EventuallyWithT(t, func(c *assert.CollectT) {
		conn, err := net.Dial("tcp", "127.0.0.1:30000")
		if assert.NoError(err) {
			return
		}
		n, err := conn.Read(buf)
		if assert.NoError(err) {
			return
		}
	}, 100*time.Millisecond, 10*time.Millisecond)
```

I would want just write:

```go
	require.EventuallyWithT(t, func(c *assert.CollectT) {
		conn, err := net.Dial("tcp", address)
		require.NoError(err)
		n, err := conn.Read(buf)
		require.NoError(err)
	}, 100*time.Millisecond, 10*time.Millisecond)
```

Reply:
I tend to use `require` and `assert` within the same context. I use `require` when I want the unexpected condition to stop the test, so here I would expect it to panic as it does normally. If I want a test to fail and retry, I would probably use the longer 
```golang
assert.EventuallyWithT(t, func(c *assert.CollectT) {
  conn, err := net.Dial("tcp", "127.0.0.1:30000")
  if assert.NoError(err) {
	  return
  }
  n, err := conn.Read(buf)
  if assert.NoError(err) {
	  return
  }
}, 100*time.Millisecond, 10*time.Millisecond)
```

Reply:
Closing this PR, thank you for the review !